src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Host/IHostTaskExecutor.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.LanguageServerProtocol.Handlers;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol
+{
+    /// <summary>
+    /// A task executor that can be used to execute LSP tasks in the host environment.
+    /// </summary>
+    public interface IHostTaskExecutor
+    {
+        /// <summary>
+        ///  Executes a task in host environment if host task executor is available.
+        ///  This is critical to trust between LSP and its hosts.
+        ///  LSP should correctly use this and wrap particulat steps that need to run in host using these methods. 
+        /// </summary>
+        /// <typeparam name="TOutput">Output Type.</typeparam>
+        /// <param name="task">Task to run inside host.</param>
+        /// <param name="operationContext">Language Server Operation Context.</param>
+        /// <param name="cancellationToken">Cancellation Token.</param>
+        /// <param name="defaultOutput">Default Output to return if the provided task is canceled by host.</param>
+        /// <returns>Output.</returns>
+        Task<TOutput> ExecuteTaskAsync<TOutput>(Func<Task<TOutput>> task, LanguageServerOperationContext operationContext, CancellationToken cancellationToken, TOutput defaultOutput = default);
+
+        /// <summary>
+        ///  Executes a task in host environment if host task executor is available.
+        ///  This is critical to trust between LSP and its hosts.
+        ///  LSP should correctly use this and wrap particulat steps that need to run in host using these methods. 
+        /// </summary>
+        /// <param name="operationContext">Language Server Operation Context.</param>
+        /// <param name="task">Task to run inside host.</param>
+        /// <param name="cancellationToken">Cancellation Token.</param>
+        Task ExecuteTaskAsync(Action task, LanguageServerOperationContext operationContext, CancellationToken cancellationToken);

```

Comment:
>Task ExecuteTaskAsync(Action task, [](http://example.com/codeflow?start=8&length=34)

Interfaces should  be minimal -  can we just make this an extension method. It's a convenience over ExecuteTask<void>

Reply:
I removed this method. Agreed interfaces should be minimal but then we also try to avoid making changes to interfaces that we define out of the fear that it would break hosts (mostly compilation errors) and derive many other interfaces. Having too many interfaces because we don't want to modify existing ones as it breaks hosts is also not ideal as it can confuse hosts on which interfaces to implement. We need to reeval this strategy at some point but for now i have gotten rid of this method